File: solution/1200-1299/1259.Handshakes That Don't Cross/Solution.java
class Solution {
    public int numberOfWays(int numPeople) {
        final int mod = (int) 1e9 + 7;
        long[] f = new long[numPeople + 1];
        f[0] = 1;
        for (int i = 2; i <= numPeople; i += 2) {
            for (int j = 0; j < i - 1; j += 2) {
                int k = i - j - 2;
                f[i] = (f[i] + f[j] * f[k]) % mod;
            }
        }
        return (int) f[numPeople];
    }
}

File: solution/1200-1299/1259.Handshakes That Don't Cross/Solution.py
class Solution:
    def numberOfWays(self, numPeople: int) -> int:
        mod = 10**9 + 7
        f = [0] * (numPeople + 1)
        f[0] = 1
        for i in range(2, numPeople + 1, 2):
            for j in range(0, i - 1, 2):
                k = i - j - 2
                f[i] = (f[i] + f[j] * f[k]) % mod
        return f[numPeople]
